Dockerfile

```diff
@@ -17,11 +17,12 @@ FROM multiarch/alpine:${ARCH}-latest-stable
 LABEL maintainer="Jan Collijs"
 
 ENV DNS1 1.1.1.1
-ENV UPSTREAM1 https://${DNS1}/.well-known/dns-query
+ENV UPSTREAM1 https://${DNS1}/dns-query
 ENV DNS2 1.0.0.1
-ENV UPSTREAM2 https://${DNS2}/.well-known/dns-query
+ENV UPSTREAM2 https://${DNS2}/dns-query
 ENV PORT 5054
 ENV ADDRESS 0.0.0.0
+ENV METRICS 127.0.0.1:8080
```

Comment:
I justed tested out this Dockerfile on my local environment, it seems to to get the metrics working on port 8080 I have to use 0.0.0.0:8080 instead of 127.0.0.1?

Reply:
Yes, I kept the default to not expose the metrics server, to expose it run the container with '-e METRICS="0.0.0.0:8080"'

All it does is adding the environment variable as option so it can be exposed and also use ip instead of hostname so it doesn't need to resolve.

> I added metrics env with a default of "127.0.0.1" so it does not need to resolve. This also allows to change the listener IP and thus expose the metrics server.

https://developers.cloudflare.com/argo-tunnel/reference/arguments default for metrics is localhost

Reply:
Makes sense! Thanks for the clarification.